src/runtime_src/core/tools/common/XBHelpMenusCore.cpp

```diff
@@ -178,10 +178,35 @@ XBUtilities::create_usage_string( const boost::program_options::options_descript
       continue;
 
     std::string completeOptionName = option->canonical_display_name(po::command_line_style::allow_dash_for_short);
+    
+    // See if we have a long flag
+    if (completeOptionName.size() != SHORT_OPTION_STRING_SIZE)
+      continue;
 
     buffer << " [" << completeOptionName << " arg]";
   }
 
+  // Gather up the options with arguments (options with no short versions)
+  for (auto & option : options) {
+    // Skip if there are no arguments
+    if (option->semantic()->max_tokens() == 0)
+      continue;
+
+    // This option shouldn't be required
+    if (option->semantic()->is_required() == true) 

```

Comment:
```suggestion
    if (option->semantic()->is_required()) 

```
I do not think the extra comparison is needed here

Reply:
extra check is ok, it makes it easier to read

Reply:
I think we should stick to just having the boolean without the comparison, I think it makes it harder to read. General coding rules say to just use the expression